packages/commonwealth/client/scripts/views/components/proposals/voting_results.tsx

```diff
@@ -21,14 +21,31 @@ import {
   YesNoRejectVotingResult,
 } from './voting_result_components';
 import useForceRerender from 'hooks/useForceRerender';
+import { IVote } from 'models/interfaces';
+import Proposal from 'models/Proposal';
+import useAaveProposalVotesQuery from 'state/api/proposals/aave/fetchAaveProposalVotes';
+import { ChainNetwork } from 'common-common/src/types';
+import useAaveProposalsQuery from 'state/api/proposals/aave/fetchAaveProposals';
 
 type VotingResultsProps = { proposal: AnyProposal };
 
 export const VotingResults = (props: VotingResultsProps) => {
   const { proposal } = props;
   const forceRerender = useForceRerender();
+  const [isLoading, setLoading] = useState(
+    !app.chain || !app.chain.loaded || !app.chain.apiInitialized
+  );
 
-  const votes = proposal.getVotes();
+  useEffect(() => {
+    app.chainAdapterReady.on('ready', () => setLoading(false));
+
+    return () => {
+      app.chainAdapterReady.off('ready', () => {
+        setLoading(false);
+        app.chainAdapterReady.removeAllListeners();
```

Comment:
do you need this? `app.chainAdapterReady.off` should be enough to remove the handler 🤔

Reply:
If you mean the `app.chainAdapterReady.removeAllListeners()` section, it is taken directly from `scripts/views/pages/proposals.tsx`. I pushed a change so that it removes only the listener that was just added.